### PR TITLE
Add config to enable deprecated APIs

### PR DESCRIPTION
**Pull request by GANGANI**
### Proposed changes in this pull request

This PR will add config to disable deprecated APIs.

The following config will have to add to the deployment.toml file to enable the /identity/connect/register API and the OAuth 1.0 API.

```
[legacy_features]
enable_oauth1 = true
enable_identity_connect_dcr = true
```

### When should this PR be merged
This PR should be merged before https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/1407

### CODE REVIEW

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 1]
```diff
@@ -314,6 +314,9 @@
   "authentication.jit_provisioning.username_provisioning_url": "/accountrecoveryendpoint/register.do",
   "authentication.jit_provisioning.password_provisioning_url": "/accountrecoveryendpoint/signup.do",
 
+  "LegacyFeatures.EnableOAuth1": false,
```
Comment (dewniMW) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:317]:
```suggestion
  "legacyFeatures.enableOAuth1": false,
```
As lowercase has been used throughout the default.json

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json [hunk 2]
```diff
@@ -314,6 +314,9 @@
...
+  "LegacyFeatures.EnableIdentityConnectDCR": false,
```
Comment (dewniMW) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json:318]: use lower case to maintain the consistency with other configs defined in default.json

Comment (GANGANI) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/org.wso2.carbon.identity.core.server.feature.default.json]: Fixed https://github.com/wso2/carbon-identity-framework/pull/2997/commits/2ce0e366e148c159060aa4ad2103ae94416682bb

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 3]
```diff
@@ -92,6 +92,11 @@
         {% endif %}
     </ServiceProviders>
 
+    <LegacyFeatures>
+        <EnableOAuth1>{{legacy_feature.enable_oauth1}}</EnableOAuth1>
```
Comment (emswbandara) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:96]: shall we make this legacy_features.enable_oauth1? and the below one too

Comment (GANGANI) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2]: Fixed https://github.com/wso2/carbon-identity-framework/pull/2997/commits/30e3885b3cab329ad0b92e2fa02e7dbfaf77c413

**Pull request closed without merging**